fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/impl/content/registry/ToolBlockTransformationRegistries.java

```
@@ -0,0 +1,86 @@
+	public static final ContentRegistry<Block, BlockState> TILLABLE_BLOCKS = new ContentRegistryImpl<>("tillable_block_registry", HoeAccessor::addTillable, HoeAccessor::removeTillable, HoeAccessor::getTilled);
+	public static final ContentRegistry<Block, BlockState> FLATTENABLE_BLOCKS = new ContentRegistryImpl<>("flattenable_block_registry", ShovelAccessor::addFlattenable, ShovelAccessor::removeFlattenable, ShovelAccessor::getFlattenable);
+	public static final ContentRegistry<Block, Block> STRIPPABLE_BLOCKS = new ContentRegistryImpl<>("strippable_block_registry", AxeAccessor::addStrippable, AxeAccessor::removeStrippable, AxeAccessor::getStrippable);
+
+	static class HoeAccessor extends HoeItem {
+		private HoeAccessor(ToolMaterial material, float attackSpeed, Settings settings) {
+			super(material, attackSpeed, settings);
+		}
+
+		static void addTillable(Block block, BlockState tilled) {
+			TILLED_BLOCKS.put(block, tilled);
+		}
+
+		static void removeTillable(Block block) {
+			TILLED_BLOCKS.remove(block);
+		}
+
+		static BlockState getTilled(Block block) {
+			return TILLED_BLOCKS.get(block);
+		}
+	}
+
+	static class ShovelAccessor extends ShovelItem {
+		private ShovelAccessor(ToolMaterial material, float attackDamage, float attackSpeed, Settings settings) {
+			super(material, attackDamage, attackSpeed, settings);
+		}
+
+		static void addFlattenable(Block block, BlockState flattened) {
+			PATH_BLOCKSTATES.put(block, flattened);
+		}
+
+		static void removeFlattenable(Block block) {
+			PATH_BLOCKSTATES.remove(block);
+		}
+
+		static BlockState getFlattenable(Block block) {
+			return PATH_BLOCKSTATES.get(block);
+		}
+	}
+
+	static class AxeAccessor extends AxeItem {
+		private AxeAccessor(ToolMaterial material, float attackDamage, float attackSpeed, Settings settings) {
+			super(material, attackDamage, attackSpeed, settings);
+		}
+
+		static void addStrippable(Block block, Block stripped) {
+			STRIPPED_BLOCKS.put(block, stripped);
```

Maybe check for the `AXIS` property here?

---

Why does it need to check for an `AXIS` property?

---

The current docs:
> Note: Both KEY and VALUE must have the {@link PillarBlock#AXIS} property!

Because it goes through vanilla's stripping system, it should throw here and crash early if that property is not present.